python/cufinufft/cufinufft/_plan.py

```diff
@@ -112,14 +112,25 @@ def __init__(self, nufft_type, n_modes, n_trans=1, eps=1e-6, isign=None,
         else:
             raise TypeError("Expected complex64 or complex128.")
 
-        if isinstance(n_modes, numbers.Integral):
-            n_modes = (n_modes,)
-        elif isinstance(n_modes, collections.abc.Iterable):
-            n_modes = tuple(n_modes)
+        if nufft_type == 3:
+            if isinstance(n_modes, numbers.Integral):
+                dim = n_modes
+                n_modes = (1,) * dim     # Ignored, can be anything
+            else:
+                raise ValueError("Type 3 plan must have n_modes_or_dim equal to the dimension")
```

Comment:
it seems that we are checking for the type but printing an error that is about the value

Reply:
Well it's a bit badly formulated. It's supposed to say that `n_modes_or_dim` must be a single number, the dimension. I'll fix it.